### ISSUE TITLE
Make Workflow Action parameters easier to work with

### ISSUE THREAD

**Issue by frikky** [frikky/Shuffle#184]
![image](https://user-images.githubusercontent.com/5719530/96943066-3b478880-14d7-11eb-8fe3-be5ddfcbb26e.png)

Workflow Variables currently have quite a few options around them. One thing that's missing is a way to do larger edits doable in an easy way. 

Example:
* You want to create a JSON object in Shuffle, but the field is so small you end up going outside Shuffle in an editor, then pasting it in. We want to fix that. One solution may be to have a "pop-out" text field that's larger and which can e.g. handle tabs and basic syntax. 

I'd think we should start by having a "fullscreen" button which makes a popout of the window, then gives you access to a JSON / code editor.

**Comment by frikky** [comment]
There are a lot of fields like this one in Shuffle. What I'd like to do is have some component that is kind of like a code editor. In some cases, people may want to write Python or Bash, as well as JSON in there, so some kind of basic syntax highlighting would be helpful.

It should also be possible to point out mistakes they've made in there, so that the frontend can do some of the verification before it goes to the backend.

TLDR: Make a code editor popout that we can put "anywhere".

**Comment by frikky** [comment]
Another thing: Make the "TextField"s into Inputs where we can highlight text with color, boldness etc. This is so that you can look at something and IMMEDIATELY know whether it's a variable, a node etc. 

Probably do this within the text editor first.

**Comment by frikky** [comment]
1. Create a code/text editor
  - Modal popout (use as component)
  - Accept image, title and subtitle as text
  - Be able to close and/or submit the modal back to the original textfield
  - Start by making it work for fields within a shuffle action
![image](https://user-images.githubusercontent.com/5719530/113562742-ee7be800-9606-11eb-842e-c817d138b1ca.png)

2. Text highlight
3. Autocompletion

**Comment by frikky** [comment]
Having something like this, with a preview of the text to be added seems like a good idea: 
![image](https://user-images.githubusercontent.com/5719530/133655696-282d053e-2dcb-46bf-b72b-92d696c706a4.png)

**Comment by frikky** [comment]
- [ ] Inline python
- [ ] Liquid Formatting (Shopify)
- [ ] JSON formatting
- [ ] Text highlighting 
- [ ] Autocompletion (button & drag&drop)

**Comment by frikky** [comment]
![image](https://user-images.githubusercontent.com/5719530/144408165-7641b91d-869a-4bf4-989e-5f5096fe8305.png)

Use the variable and check whether it exists. Use values from ParsedAction.jsx as they are being used in this dropdown for previous actions:

![image](https://user-images.githubusercontent.com/5719530/144408294-7fc81755-64d0-4861-9277-91270c8532a2.png)

**Comment by frikky** [comment]
I wonder whether we should add something like this as well. Favorite parts:
* Basic validation Functions
* Math
* Array handlers
* Text manipulation
* Timestamp functions

![image](https://user-images.githubusercontent.com/5719530/150700325-374bfe1a-5cd6-4c3c-9fab-d98d2e414312.png)

PS: This is from Integromat.

**Comment by frikky** [comment]
We're on the right track. Continuing from May 15th
